src/jsifier.mjs

```diff
@@ -79,6 +80,17 @@ function stringifyWithFunctions(obj) {
   if (Array.isArray(obj)) {
     return '[' + obj.map(stringifyWithFunctions).join(',') + ']';
   }
+
+  // preserve the type of the object if it is one of [Map, Set, WeakMap, WeakSet].
+  const builtinContainers = vm.runInContext('[Map, Set, WeakMap, WeakSet]', compileTimeContext);
```

Comment:
Sorry to go back on this again, but can we use `runInMacroContext('...', {filename: '<internal>'})` and avoid importing `vm`?